app/react/Templates/components/specs/MetadataTemplate.spec.js

```diff
@@ -228,21 +230,49 @@ describe('MetadataTemplate', () => {
       });
     });
 
-    describe('when the mapping has con flicts', () => {
-      it('should ask for a reindex', async () => {
+    describe('confirmation of saving', () => {
+      let context;
+      const templateWithId = {
+        _id: 'template1',
+        properties: [{ name: 'dob', type: 'date', label: 'Date of birth' }],
+      };
+
+      async function submitTemplate(templateToSubmit, validMapping = true, entityCount = 100) {
+        context = {
+          confirm: jasmine.createSpy('confirm'),
+        };
         spyOn(api, 'validateMapping').and.returnValue({
           error: 'error',
-          valid: false,
+          valid: validMapping,
         });
-        const context = { confirm: jasmine.createSpy('confirm') };
-
+        spyOn(entitiesApi, 'countByTemplate').and.returnValue(entityCount);
         const component = shallow(<MetadataTemplate {...props} />, { context });
-        const template = { properties: [{ name: 'dob', type: 'date', label: 'Date of birth' }] };
-        await component.instance().onSubmit(template);
-        context.confirm.calls.mostRecent().args[0].accept();
-        expect(props.saveTemplate).toHaveBeenCalledWith({
-          properties: [{ name: 'dob', type: 'date', label: 'Date of birth' }],
-          reindex: true,
+        await component.instance().onSubmit(templateToSubmit);
+      }
+
+      describe('when the mapping has conflicts', () => {
+        it('should ask for a reindex', async () => {
+          await submitTemplate(templateWithId, false);
+          context.confirm.calls.mostRecent().args[0].accept();
+          expect(props.saveTemplate).toHaveBeenCalledWith({
+            ...templateWithId,
+            reindex: true,
+          });
+        });
+
+        describe('when there is a quite amount of entities from the template', () => {
+          it('should ask for a reindex', async () => {
+            await submitTemplate(templateWithId, true, 50000);
+            context.confirm.calls.mostRecent().args[0].cancel();
```

Comment:
Could we add a test case to check when the user click yes?

Reply:
done